apps/synapse-oauth-signin/src/AppInitializer.tsx

```diff
@@ -81,47 +53,36 @@ function AppInitializer(
           )}?${urlSearchParams.toString()}`,
         )
       })
-    } else {
-      SynapseClient.getAccessTokenFromCookie()
-        .then((accessToken: string | null) => {
-          if (accessToken) {
-            // check max age when re-establishing the session, not to auto-consent.
-            const maxAgeURLParam = urlSearchParams.get('max_age')
-            // SWC-5597: if max_age is defined, then return if the user last authenticated more than max_age seconds ago
-            if (maxAgeURLParam && parseInt(maxAgeURLParam)) {
-              SynapseClient.getAuthenticatedOn(accessToken).then(
-                (authenticatedOnResponse: AuthenticatedOn) => {
-                  const lastAuthenticatedOn = moment.utc(
-                    authenticatedOnResponse.authenticatedOn,
-                  )
-                  const now = moment.utc()
-                  if (
-                    now.diff(lastAuthenticatedOn, 'seconds') <=
-                    parseInt(maxAgeURLParam)
-                  )
-                    setAccessToken(accessToken)
-                },
-              )
-            } else {
-              // no max age param, use the token
-              setAccessToken(accessToken)
-            }
-          }
-        })
-        .catch(_err => {
-          console.log('no token from cookie could be fetched ', _err)
-          if (prompt === 'none') {
-            // not logged in, and prompt is "none".
-            handleErrorRedirect(
-              new OAuthClientError(
-                'login_required',
-                'User is not logged in, and prompt was set to none',
-              ),
-            )
+    }
+  }, [])
+
+  useEffect(() => {
+    const urlSearchParams = new URLSearchParams(window.location.search)
+    SynapseClient.getAccessTokenFromCookie().then(
+      (accessToken: string | null) => {
+        if (accessToken) {
+          // check max age when re-establishing the session, not to auto-consent.
+          const maxAgeURLParam = urlSearchParams.get('max_age')
+          // SWC-5597: if max_age is defined, then return if the user last authenticated more than max_age seconds ago
+          if (maxAgeURLParam && parseInt(maxAgeURLParam)) {
+            setMaxAge(parseInt(maxAgeURLParam))
```

Comment:
This `max_age` value is passed via query parameter in the URL that an OAuth client app will use. The logic that decides how that impacts login state is now handled in the `ApplicationSessionManager`, so `maxAge` is now just passed to `ApplicationSessionManager` as a prop.